### PR TITLE
fix(button): add the focus() method to buttons.

### LINKED ISSUE #510: focus event does not work
1. the readme example for de sidenav component  does not work at all.

<md-sidenav-layout class="demo-root" fullscreen>
  <md-sidenav #start (open)="mybutton.focus()">
      Start Sidenav.
      <br>
      <button md-button #mybutton (click)="start.close()">Close</button>
    </md-sidenav>
    <md-sidenav #end align="end">
      End Sidenav.
      <button md-button (click)="end.close()">Close</button>
    </md-sidenav>
    My regular content. This will be moved into the proper DOM at runtime.
  </md-sidenav-layout>

2. I made a little change

<md-sidenav-layout class="demo-root" fullscreen>
   <md-sidenav #start (open)="mybutton.focus()">
      Start Sidenav.
    </md-sidenav>
    <md-sidenav #end align="end">
      End Sidenav.
      <button md-button (click)="end.close()">Close</button>
    </md-sidenav>
    <br>
      <button md-button #mybutton (click)="start.open()">Open</button>
    My regular content. This will be moved into the proper DOM at runtime.
</md-sidenav-layout>

And when the sidenave was opened, the following error as showing:

EXCEPTION: Error in demo-app/demo-app.html:1:22
browser_adapter.js:77 EXCEPTION: Error in demo-app/demo-app.html:1:22BrowserDomAdapter.logError @ browser_adapter.js:77BrowserDomAdapter.logGroup @ browser_adapter.js:87ExceptionHandler.call @ exception_handler.js:57(anonymous function) @ application_ref.js:265schedulerFn @ async.js:123SafeSubscriber.__tryOrUnsub @ Subscriber.js:225SafeSubscriber.next @ Subscriber.js:174Subscriber._next @ Subscriber.js:124Subscriber.next @ Subscriber.js:88Subject._finalNext @ Subject.js:128Subject._next @ Subject.js:120Subject.next @ Subject.js:77EventEmitter.emit @ async.js:112NgZone._zoneImpl.ng_zone_impl_1.NgZoneImpl.onError @ ng_zone.js:120NgZoneImpl.inner.inner.fork.onHandleError @ ng_zone_impl.js:66ZoneDelegate.handleError @ zone.js:327Zone.runTask @ zone.js:259ZoneTask.invoke @ zone.js:423
browser_adapter.js:77 ORIGINAL EXCEPTION: TypeError: self._MdButton_13_4.focus is not a functionBrowserDomAdapter.logError @ browser_adapter.js:77ExceptionHandler.call @ exception_handler.js:66(anonymous function) @ application_ref.js:265schedulerFn @ async.js:123SafeSubscriber.__tryOrUnsub @ Subscriber.js:225SafeSubscriber.next @ Subscriber.js:174Subscriber._next @ Subscriber.js:124Subscriber.next @ Subscriber.js:88Subject._finalNext @ Subject.js:128Subject._next @ Subject.js:120Subject.next @ Subject.js:77EventEmitter.emit @ async.js:112NgZone._zoneImpl.ng_zone_impl_1.NgZoneImpl.onError @ ng_zone.js:120NgZoneImpl.inner.inner.fork.onHandleError @ ng_zone_impl.js:66ZoneDelegate.handleError @ zone.js:327Zone.runTask @ zone.js:259ZoneTask.invoke @ zone.js:423
browser_adapter.js:77 ORIGINAL STACKTRACE:BrowserDomAdapter.logError @ browser_adapter.js:77ExceptionHandler.call @ exception_handler.js:69(anonymous function) @ application_ref.js:265schedulerFn @ async.js:123SafeSubscriber.__tryOrUnsub @ Subscriber.js:225SafeSubscriber.next @ Subscriber.js:174Subscriber._next @ Subscriber.js:124Subscriber.next @ Subscriber.js:88Subject._finalNext @ Subject.js:128Subject._next @ Subject.js:120Subject.next @ Subject.js:77EventEmitter.emit @ async.js:112NgZone._zoneImpl.ng_zone_impl_1.NgZoneImpl.onError @ ng_zone.js:120NgZoneImpl.inner.inner.fork.onHandleError @ ng_zone_impl.js:66ZoneDelegate.handleError @ zone.js:327Zone.runTask @ zone.js:259ZoneTask.invoke @ zone.js:423
browser_adapter.js:77 TypeError: self._MdButton_13_4.focus is not a function
    at DebugAppView._View_DemoApp0._handle_open_2_0 (DemoApp.template.js:318)
    at eval (view.js:316)
    at eval (async.js:134)
    at ZoneDelegate.invokeTask (zone.js:356)
    at Object.NgZoneImpl.inner.inner.fork.onInvokeTask (ng_zone_impl.js:36)
    at ZoneDelegate.invokeTask (zone.js:355)
    at Zone.runTask (zone.js:256)
    at ZoneTask.invoke (zone.js:423)BrowserDomAdapter.logError @ browser_adapter.js:77ExceptionHandler.call @ exception_handler.js:70(anonymous function) @ application_ref.js:265schedulerFn @ async.js:123SafeSubscriber.__tryOrUnsub @ Subscriber.js:225SafeSubscriber.next @ Subscriber.js:174Subscriber._next @ Subscriber.js:124Subscriber.next @ Subscriber.js:88Subject._finalNext @ Subject.js:128Subject._next @ Subject.js:120Subject.next @ Subject.js:77EventEmitter.emit @ async.js:112NgZone._zoneImpl.ng_zone_impl_1.NgZoneImpl.onError @ ng_zone.js:120NgZoneImpl.inner.inner.fork.onHandleError @ ng_zone_impl.js:66ZoneDelegate.handleError @ zone.js:327Zone.runTask @ zone.js:259ZoneTask.invoke @ zone.js:423
browser_adapter.js:77 ERROR CONTEXT:BrowserDomAdapter.logError @ browser_adapter.js:77ExceptionHandler.call @ exception_handler.js:73(anonymous function) @ application_ref.js:265schedulerFn @ async.js:123SafeSubscriber.__tryOrUnsub @ Subscriber.js:225SafeSubscriber.next @ Subscriber.js:174Subscriber._next @ Subscriber.js:124Subscriber.next @ Subscriber.js:88Subject._finalNext @ Subject.js:128Subject._next @ Subject.js:120Subject.next @ Subject.js:77EventEmitter.emit @ async.js:112NgZone._zoneImpl.ng_zone_impl_1.NgZoneImpl.onError @ ng_zone.js:120NgZoneImpl.inner.inner.fork.onHandleError @ ng_zone_impl.js:66ZoneDelegate.handleError @ zone.js:327Zone.runTask @ zone.js:259ZoneTask.invoke @ zone.js:423
browser_adapter.js:77DebugContextBrowserDomAdapter.logError @ browser_adapter.js:77ExceptionHandler.call @ exception_handler.js:74(anonymous function) @ application_ref.js:265schedulerFn @ async.js:123SafeSubscriber.__tryOrUnsub @ Subscriber.js:225SafeSubscriber.next @ Subscriber.js:174Subscriber._next @ Subscriber.js:124Subscriber.next @ Subscriber.js:88Subject._finalNext @ Subject.js:128Subject._next @ Subject.js:120Subject.next @ Subject.js:77EventEmitter.emit @ async.js:112NgZone._zoneImpl.ng_zone_impl_1.NgZoneImpl.onError @ ng_zone.js:120NgZoneImpl.inner.inner.fork.onHandleError @ ng_zone_impl.js:66ZoneDelegate.handleError @ zone.js:327Zone.runTask @ zone.js:259ZoneTask.invoke @ zone.js:423
Subscriber.js:229 Uncaught EXCEPTION: Error in demo-app/demo-app.html:1:22
ORIGINAL EXCEPTION: TypeError: self._MdButton_13_4.focus is not a function
ORIGINAL STACKTRACE:
TypeError: self._MdButton_13_4.focus is not a function
    at DebugAppView._View_DemoApp0._handle_open_2_0 (DemoApp.template.js:318:35)
    at eval (http://localhost:4200/vendor/@angular/core/src/linker/view.js:316:24)
    at eval (http://localhost:4200/vendor/@angular/core/src/facade/async.js:134:94)
    at ZoneDelegate.invokeTask (http://localhost:4200/vendor/zone.js/dist/zone.js:356:38)
    at Object.NgZoneImpl.inner.inner.fork.onInvokeTask (http://localhost:4200/vendor/@angular/core/src/zone/ng_zone_impl.js:36:41)
    at ZoneDelegate.invokeTask (http://localhost:4200/vendor/zone.js/dist/zone.js:355:43)
    at Zone.runTask (http://localhost:4200/vendor/zone.js/dist/zone.js:256:48)
    at ZoneTask.invoke (http://localhost:4200/vendor/zone.js/dist/zone.js:423:34)
ERROR CONTEXT:
[object Object]

### PR DESCRIPTION
**Pull request by hansl**
Fixes #510.

### CODE REVIEW

**Comment by hansl** [comment]
R: @jelbourn @robertmesserle @kara

File: src/components/button/button.ts [hunk 1]
```diff
@@ -80,6 +80,10 @@ export class MdButton {
   removeKeyboardFocus() {
     this.isKeyboardFocused = false;
   }
+
+  focus() {
+    this.elementRef.nativeElement.focus();
+  }
```
Comment (jelbourn) [inline src/components/button/button.ts]: Add a `TODO` to add an `e2e` test for this? (simulating focus in a unit test doesn't really work well)

Comment (hansl) [inline src/components/button/button.ts]: done.

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**